modules/apps/remote-app/remote-app-service/src/main/java/com/liferay/remote/app/service/impl/RemoteAppEntryLocalServiceImpl.java

```diff
@@ -106,11 +125,18 @@ public RemoteAppEntry addCustomElementRemoteAppEntry(
 
 		_validateFriendlyURLMapping(friendlyURLMapping);
 
-		RemoteAppEntry remoteAppEntry = remoteAppEntryPersistence.create(
-			counterLocalService.increment());
+		if (Validator.isBlank(externalReferenceCode)) {
```

Comment:
Usually we have a default externalReferenceCode when it is null or empty, that is the entity id as string. I think it is useful to avoid having null external reference codes in the database.

You should apply it also in the database upgrade process